pkg/solana/txm/pendingtx.go

```diff
@@ -44,14 +46,21 @@ type PendingTxContext interface {
 	TrimFinalizedErroredTxs()
 }
 
-type pendingTx struct {
-	tx          solana.Transaction
-	cfg         TxConfig
-	signatures  []solana.Signature
-	id          string
-	createTs    time.Time
-	retentionTs time.Time
-	state       TxState
+type PendingTx struct {
+	Tx                   solana.Transaction
+	AccountID            string
+	cfg                  TxConfig
+	signatures           []solana.Signature
+	UUID                 string
```

Comment:
I think this name might be misleading. This won't necessarily be a uuid. We allow the caller to provide a unique identifier for a tx so they can track it's status externally. Some of them like CCIP might make it some deterministic string that they can build from their metadata so they don't have to store a uuid.